File: config/rules/opengrep/python/flask/cwe-614-insecure-cookie/insecure-cookie.py
from flask import make_response

response = make_response("OK")

# ruleid: python.flask.security.cwe-614.insecure-cookie
response.set_cookie("session_id", value=token, secure=False)

# ruleid: python.flask.security.cwe-614.insecure-cookie
response.set_cookie("session_id", value=token, httponly=False)

# ruleid: python.flask.security.cwe-614.insecure-cookie
SESSION_COOKIE_SECURE = False

# ok: python.flask.security.cwe-614.insecure-cookie
response.set_cookie("session_id", value=token, secure=True, httponly=True)

# ok: python.flask.security.cwe-614.insecure-cookie
SESSION_COOKIE_SECURE = True

File: config/rules/opengrep/csharp/aspnet-core/cwe-614-insecure-cookie/insecure-cookie.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Authentication.Cookies;

public class InsecureCookieExamples
{
    public void Vulnerable(HttpResponse response)
    {
        // ruleid: csharp.aspnet.security.cwe-614.insecure-cookie
        var options = new CookieOptions { Secure = false, HttpOnly = true };
        response.Cookies.Append("session", token, options);

        // ruleid: csharp.aspnet.security.cwe-614.insecure-cookie
        var opts2 = new CookieOptions { Secure = true, HttpOnly = false };

        // ruleid: csharp.aspnet.security.cwe-614.insecure-cookie
        options.Secure = false;

        // ruleid: csharp.aspnet.security.cwe-614.insecure-cookie
        options.Cookie.SecurePolicy = CookieSecurePolicy.None;
    }

    public void Safe(HttpResponse response)
    {
        // ok: csharp.aspnet.security.cwe-614.insecure-cookie
        var options = new CookieOptions { Secure = true, HttpOnly = true };
        response.Cookies.Append("session", token, options);

        // ok: csharp.aspnet.security.cwe-614.insecure-cookie
        options.Cookie.SecurePolicy = CookieSecurePolicy.Always;
    }
}
